test/dynamo/test_functions.py

```diff
@@ -151,6 +151,22 @@ def test_inline_script_if_tracing_fn_with_default_args(a, b):
     def test_inline_lru_cache_fn_with_default_args(a, b):
         return inline_lru_cache_fn_with_default_args(a, 2, b)
 
+
+    def test_lru_cache_warning_issued_during_tracing(self):
+        from functools import lru_cache
+        import warnings
+
+        @lru_cache
+        def foo(x):
+            return x + 1
+
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter("always")
+            torch.compile(foo)(torch.randn(4))
+            self.assertEqual(len(w), 1)
```

Comment:
Don't check for just one message - warnings might be emitted elsewhere as well. Just iterate through all warnings to look one that comes from your added warning.